sphinx/transforms/post_transforms/images.py

```diff
@@ -187,16 +194,14 @@ class ImageConverter(BaseImageConverter):
     conversion_rules = []  # type: List[Tuple[str, str]]
 
     def __init__(self, *args: Any, **kwargs: Any) -> None:
-        self.available = None   # type: bool
-                                # the converter is available or not.
-                                # Will be checked at first conversion
         super().__init__(*args, **kwargs)
 
     def match(self, node: nodes.image) -> bool:
         if not self.app.builder.supported_image_types:
             return False
         elif self.available is None:
-            self.available = self.is_available()
+            # store the value to the class variable to share it during the build
+            self.__class__.available = self.is_available()
```

Comment:
How about using [functools.lru_cache](https://docs.python.org/3/library/functools.html#functools.lru_cache) ?

```
>>> import functools
>>> import random
>>> class F:
...   @functools.lru_cache
...   def rand(self):
...     return random.randint(1,10)
...
>>> f = F()
>>> f.rand()
9
>>> f.rand()
9
>>> f2 = F()
>>> f2.rand()
9
>>> f3 = F()
>>> f3.rand()
9
```

Reply:
Good point. If we choose `lru_cache()`, it is recommended to all subclasses to wrap `is_available()` method by `lru_cache()`. But it's natural design to me. So +0 for your proposal.

Reply:
I tested `functools.lru_cache()`. But it does not share the cache between instances. I guess different first argument `self` is passed on each instance.
```
$ python
Python 3.8.6 (default, Sep 27 2020, 22:28:24)
[Clang 11.0.3 (clang-1103.0.32.59)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import functools
>>> import random
>>> class F:
...   @functools.lru_cache
...   def rand(self):
...     return random.randint(1,10)
...
>>> f = F()
>>> f.rand()
8
>>> f.rand()
8
>>> f2 = F()
>>> f2.rand()
7
>>> f3 = F()
>>> f3.rand()
7
>>> f4 = F()
>>> f4.rand()
5
```